documentation/src/main/asciidoc/topics/con_performance_clustered_execution.adoc

```diff
@@ -0,0 +1,56 @@
+[id='performance-clustered-execution_{context}']
+= Code execution and data processing
+
+One of the benefits of distributed caching is that you can leverage compute resources from each host to perform large scale data processing more efficiently.
+By executing your processing logic directly on {brandname} you spread the workload across multiple Java Virtual Machines (JVMs).
+Your code also runs in the same memory space where {brandname} stores your data, meaning that you can iterate over millions of entries per millisecond.
+// Review comment
+// Will: Does this sound correct?
+// Should we make it more generic or s/millions/billions?
+
+In terms of performance impact to your {brandname} deployment, that entirely depends on your code execution.
+More complex processing operations have higher performance penalties so you should approach running any code on {brandname} clusters with careful planning.
+Start out by testing your code and performing multiple execution runs on a smaller, sample data set.
+After you gather some metrics you can start identifying optimizations and understanding what performance implications of the code you're running.
+
+One definite consideration is that long running processes can start having a negative impact on normal read and write operations.
+So it is imperative that you monitor your deployment over time and continually assess performance.
+
+[discrete]
+== Embedded caches
+
+With embedded caches, {brandname} provides a `ClusterExecutor` API that lets you execute code right next to the data and directly on the cluster.
+You can execute your code on all nodes in the {brandname} or confine execution to primary owners.
+
+While the `ClusterExecutor` API lets you iterate over the entire data set, you might prefer to perform your operations on collections instead.
+To achieve this you use the `Stream` API.
+
+Another way to think about the distinction between the `ClusterExecutor` API and the `Stream` API is that clustered execution gives you processing based on {brandname} nodes and distributed streams give you processing based on data.
+Distributed streams also has the advantage of adjusting to cluster topology changes.
+For clustered execution, when nodes drop out of the cluster...
```

Comment:
not sure whether this ... is intentional, at least it is not helpful and we should clarify the consequences

Reply:
Indeed it is intentional. I've added a review comment for Will just underneath the ellipsis and still need to follow up with him. Nice catch though.